pbj-core/pbj-compiler/src/main/java/com/hedera/pbj/compiler/impl/generators/ModelGenerator.java

```diff
@@ -189,14 +192,50 @@ public void generate(
                 .indent(DEFAULT_INDENT);
         bodyContent += "\n";
 
-        // constructor
-        bodyContent += generateConstructor(javaRecordName, fields, fieldsNoPrecomputed, true, msgDef, lookupHelper);
+        bodyContent += "private final List<UnknownField> $unknownFields;".indent(DEFAULT_INDENT);
+        bodyContent += "\n";
+        bodyContent += "\n";
+
+        // constructors: w/o unknownFields, and with unknownFields
+        bodyContent +=
+                generateConstructor(javaRecordName, fields, false, fieldsNoPrecomputed, true, msgDef, lookupHelper);
+        bodyContent += "\n";
+        bodyContent +=
+                generateConstructor(javaRecordName, fields, true, fieldsNoPrecomputed, true, msgDef, lookupHelper);
         bodyContent += "\n";
 
         // record style getters
         bodyContent += generateRecordStyleGetters(fieldsNoPrecomputed);
         bodyContent += "\n";
 
+        bodyContent +=
+                """
+                /**
+                 * Get an unmodifiable list of all unknown fields parsed from the original data, i.e. the fields
+                 * that are unknown to the .proto model which generated this Java model class. The fields are sorted
+                 * by their field numbers in an increasing order.
+                 * <p>
+                 * Note that by default, PBJ Codec discards unknown fields for performance reasons.
+                 * The parse() method has to be invoked with `parseUnknownFields = true` in order to populate the
+                 * unknown fields.
+                 * <p>
+                 * Also note that there may be multiple `UnknownField` items with the same field number
+                 * in case a repeated field uses the unpacked wire format. It's up to the application
+                 * to interpret these unknown fields correctly if necessary.
+                 * <p>
+                 * If the parsing of unknown fields was enabled when this model instance was parsed originally and
+                 * the unknown fields were present, then a subsequent `Codec.write()` call will persist all the parsed
+                 * unknown fields.
+                 *
+                 * @return a (potentially empty) list of unknown fields
+                 */
+                public @NonNull List<UnknownField> getUnknownFields() {
+                    return $unknownFields == null ? Collections.EMPTY_LIST : $unknownFields;
```

Comment:
The doc says the method returns an unmodifiable list. Here it returns the field as is, and it looks like the list is created as modifiable in the constructor